### ISSUE TITLE
CIFAR-10 RCAE Training and Testing with same data?

### ISSUE THREAD

**Issue by davidsjohnson** [raghavchalapathy/oc-nn#8]
Hi, 
I've been trying to implement your work for a project I am working on. While looking to implement my own `load_data()` function, I noticed it looks like you are testing your models with data that is also in the training set.  But maybe I am missing something.

In cifar10.py `load_data()` lines 147 and 148 there is the code:

`self._X_test = x_train`
`self._y_test = y_train`

Additionally in RCAE `fit_and_predict()` lines 452 and 453:

`X_test = X_train`
`y_test = y_train`

Then you pass X_test to your model's predict function

Can you please explain where the test data is being split from the training data? Additionally, why does X_train contain both anomaly and normal examples? In cifar10.py, you're concatenating x_norm with x_outlier for x_train, but I don't see where there the data is actually split training and testing.  

Thanks!

**Comment by raghavchalapathy** [comment]
Can you please explain where the test data is being split from the training data? 

These are lines where outliers are included for traning.
        outliers = list(range(0, 10))
        outliers.remove(Cfg.cifar10_normal)
        idx_outlier = np.any(y_train[..., None] == np.array(outliers)[None, ...], axis=1)
        x_outlier = x_train[idx_outlier]   

Why does X_train contain both anomaly and normal examples?
you're concatenating x_norm with x_outlier for x_train, but I don't see where there the data is actually split training and testing.

When you are training , in reality you dont know the labels of anomalies, this condition is simulated using both anomaly ( in small proportion) and normal samples .
Fit_predict is similar to OCSVM fit_predict behaviour

**Comment by davidsjohnson** [comment]
Thanks for the quick response.

Yes, I did see that the files were split into x_norm and x_outlier, but then these are concatentated together to x_train in line 138. My question, is more related to where are you splitting x_train into separate training and testing sets? Since we want to ensure that we test the models with data that the model hasn't seen in training.

**Comment by raghavchalapathy** [comment]
My question, is more related to where are you splitting x_train into
separate training and testing sets? Since we want to ensure that we test
the models with data that the model hasn't seen in training.

—I have tested both using train data and test data, you can use cross
validation too to make sure the model generalizes if you wish to

**Comment by raghavchalapathy** [comment]
Closing this issue due to inactivity
For detailed explanations please refer to the link below
https://raghavchalapathy.github.io/KDD-Tutorials-2020-Deep-Robust-Anomaly-Detection/
